providers/redfish/redfish.go

```diff
@@ -156,8 +159,30 @@ func (c *Conn) Compatible(ctx context.Context) bool {
 	return err == nil
 }
 
+func (c *Conn) sessionActive(ctx context.Context) error {
+	// skip session active checks for tests
+	if os.Getenv(internal.EnvBmclibTestActive) == "true" {
```

Comment:
any chance the `c.conn` can be replaced with a test implementation? This would remove the need for a global and test variable here. A global and test only variable living here means parallel tests are almost certainly out of the question.

Reply:
Thanks for pointing that out, I've pushed a change that mocks the Redfish sessions endpoint instead.

Reply:
To a similar tune, if you replaced `c.conn` with an unexported interface you could swap it for a construct that deals with returning the error instead. I.e. this is a problem created by the fact `Close()` sets `c.conn` to `nil` and if it instead set `c.conn = sessionlessImplementation{}` we could avoid all the `c.sessionActive()` checks.